### PR TITLE
fix: create readonly user for non-core aether modules

### PR DESCRIPTION
**Pull request by quesebifurcan**
Create readonly user in kernel database. This role can be used
by non-core Aether modules such as `aether-producer`.

The user is granted read permissions for the following tables:
- kernel_entity
- kernel_mapping
- kernel_projectschema
- kernel_schema

NOTE:
The main script is in python, not pure SQL. I tried the latter, but
there were simply too many variables to interpolate; quoting gets 
tricky when the same variable has to serve as both identifier and 
literal.

Partly resolves https://jira.ehealthafrica.org/browse/AET-315

### CODE REVIEW

File: helm/kernel/templates/deployment.yaml [hunk 1]
```diff
@@ -135,6 +135,16 @@ spec:
             secretKeyRef:
               name: {{ .Values.application.secret }}
               key: kernel-database-name
+        - name: KERNEL_READONLY_DB_USERNAME
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.application.secret }}
+              key: kernel-readonly-db-username
```
Comment (tooxie) [inline helm/kernel/templates/deployment.yaml]: I know this is consistent with the current state of the chart, but in #280 we are moving all non-password values from the secret to the values file.

**Review by tooxie — CHANGES_REQUESTED** [review]
Yayyy for the principle of least privilege!! \o/ 😄 🎉

File: helm/kernel/templates/deployment.yaml (hunk 1 above)
Comment (quesebifurcan) [reply helm/kernel/templates/deployment.yaml]: ah, that makes sense. Never seemed like a secret to me, this one.

**Review by shawnsarwar — DISMISSED** [review]

**Review by tooxie — APPROVED** [review]

File: aether-kernel/sql/create_readonly_user.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+def main():
+    dbname = os.environ['DB_NAME']
+    host = os.environ['PGHOST']
+    user = os.environ['PGUSER']
+
+    postgres_credentials = {
+        'dbname': dbname,
+        'host': host,
+        'port': 5432,
```
Comment (tooxie) [inline aether-kernel/sql/create_readonly_user.py]: Please get this from the environment as well.

**Review by tooxie — CHANGES_REQUESTED** [review]

File: aether-kernel/sql/create_readonly_user.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+BEGIN
+  IF NOT EXISTS (SELECT FROM pg_catalog.pg_roles WHERE rolname = {rolename})
+  THEN
+      CREATE ROLE {role} WITH LOGIN ENCRYPTED PASSWORD {password}
+      INHERIT NOSUPERUSER NOCREATEDB NOCREATEROLE NOREPLICATION;
+  END IF;
+END
+$$ LANGUAGE plpgsql;
+
+GRANT CONNECT ON DATABASE {database} TO {role};
```
Comment (obdulia-losantos) [inline aether-kernel/sql/create_readonly_user.py]: Could you include this instruction before the GRANT ones? Just to remove previous privileges added manually.

```SQL
REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA public FROM {role};
```

Comment (obdulia-losantos) [reply aether-kernel/sql/create_readonly_user.py]: Better 

```sql
REVOKE  ALL PRIVILEGES ON DATABASE {database} FROM {role} CASCADE;
```

https://www.postgresql.org/docs/9.6/static/sql-revoke.html

File: aether-kernel/sql/create_readonly_user.py [hunk 4]
```diff
@@ -44,6 +44,8 @@
 END
 $$ LANGUAGE plpgsql;
 
+REVOKE ALL PRIVILEGES ON DATABASE {database} FROM {role} CASCADE;
```
Comment (obdulia-losantos) [inline aether-kernel/sql/create_readonly_user.py]: :+1:

**Review by obdulia-losantos — APPROVED** [review]

**Review by tooxie — APPROVED** [review]

**Pull request merged**